addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/parserapi/ParserAPI.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrules.parserapi;
+
+import java.io.IOException;
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.Source;
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.Token;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.ascanrules.parserapi.impl.JavaScriptLexer;
+import org.zaproxy.zap.extension.ascanrules.parserapi.impl.JavaScriptParser;
+
+class MyErrorListener extends BaseErrorListener {
+    @Override
+    public void syntaxError(
+            Recognizer<?, ?> recognizer,
+            Object offendingSymbol,
+            int line,
+            int charPositionInLine,
+            String msg,
+            RecognitionException e) {
+        throw new java.lang.Error("Syntax Error");
+    }
+}
+
+public class ParserAPI {
+
+    private String scriptCode;
+
+    public ParserAPI(HttpMessage msg) {
+        String htmlCode = msg.getResponseBody().toString();
+        Source htmlSrc = new Source(htmlCode);
+        Element script = htmlSrc.getFirstElement(HTMLElementName.SCRIPT);
```

Comment:
What if it's not in the first element? Would be good to have tests.

Reply:
Now it works even if it has multiple blocks. I have also added a test.